authenticator.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"bytes"
+	"encoding/hex"
+	"io/ioutil"
+	"net/http"
+
+	p2pcrypto "github.com/libp2p/go-libp2p-crypto"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type Authenticator interface {
+	AuthenticateRequest(r *http.Request) bool
+}
+
+type P2PAuthenticator struct {
+	// B64 encoded public keys of identities whose logs are allowed to be
+	// ingested
+	allowedPubKeys []string
+
+	pubKeyMap map[string]p2pcrypto.PubKey
+}
+
+func NewP2PAuthenticator(allowedPubKeys []string) (*P2PAuthenticator, error) {
+	pubKeys := make(map[string]p2pcrypto.PubKey)
```

Comment:
Nit: 

```suggestion
	pubKeys := map[string]p2pcrypto.PubKey{}
```

Is a nice short-hand